scene/2d/audio_stream_player_2d.cpp

```diff
@@ -169,116 +64,124 @@ void AudioStreamPlayer2D::_notification(int p_what) {
-			int new_output_count = 0;
+		//stop playing if no longer active
+		if (active.is_set() && !AudioServer::get_singleton()->get_is_playback_active(stream_playback)) {
+			active.clear();
+			set_physics_process_internal(false);
+			emit_signal(SNAME("finished"));
+		}
+	}
+}
+StringName AudioStreamPlayer2D::_get_actual_bus() {
+	if (!stream_playback.is_valid()) {
+		return "Master";
+	}
 
-			Vector2 global_pos = get_global_position();
+	Vector2 global_pos = get_global_position();
 
-			int bus_index = AudioServer::get_singleton()->thread_find_bus_index(bus);
+	//check if any area is diverting sound into a bus
+	Ref<World2D> world_2d = get_world_2d();
+	ERR_FAIL_COND_V(world_2d.is_null(), "Master");
 
-			//check if any area is diverting sound into a bus
+	PhysicsDirectSpaceState2D *space_state = PhysicsServer2D::get_singleton()->space_get_direct_state(world_2d->get_space());
+	PhysicsDirectSpaceState2D::ShapeResult sr[MAX_INTERSECT_AREAS];
 
-			PhysicsDirectSpaceState2D *space_state = PhysicsServer2D::get_singleton()->space_get_direct_state(world_2d->get_space());
+	int areas = space_state->intersect_point(global_pos, sr, MAX_INTERSECT_AREAS, Set<RID>(), area_mask, false, true);
 
-			PhysicsDirectSpaceState2D::ShapeResult sr[MAX_INTERSECT_AREAS];
+	for (int i = 0; i < areas; i++) {
+		Area2D *area2d = Object::cast_to<Area2D>(sr[i].collider);
+		if (!area2d) {
+			continue;
+		}
 
-			int areas = space_state->intersect_point(global_pos, sr, MAX_INTERSECT_AREAS, Set<RID>(), area_mask, false, true);
+		if (!area2d->is_overriding_audio_bus()) {
+			continue;
+		}
 
-			for (int i = 0; i < areas; i++) {
-				Area2D *area2d = Object::cast_to<Area2D>(sr[i].collider);
-				if (!area2d) {
-					continue;
-				}
+		return area2d->get_audio_bus_name();
+	}
+	return default_bus;
+}
 
-				if (!area2d->is_overriding_audio_bus()) {
-					continue;
-				}
+void AudioStreamPlayer2D::_update_panning() {
+	if (!stream_playback.is_valid()) {
+		return;
+	}
 
-				StringName bus_name = area2d->get_audio_bus_name();
-				bus_index = AudioServer::get_singleton()->thread_find_bus_index(bus_name);
-				break;
-			}
+	last_mix_count = AudioServer::get_singleton()->get_mix_count();
 
-			const Set<Viewport *> viewports = world_2d->get_viewports();
+	Ref<World2D> world_2d = get_world_2d();
+	ERR_FAIL_COND(world_2d.is_null());
 
-			for (Set<Viewport *>::Element *E = viewports.front(); E; E = E->next()) {
-				Viewport *vp = E->get();
-				if (vp->is_audio_listener_2d()) {
-					//compute matrix to convert to screen
-					Transform2D to_screen = vp->get_global_canvas_transform() * vp->get_canvas_transform();
-					Vector2 screen_size = vp->get_visible_rect().size;
+	Vector2 global_pos = get_global_position();
 
-					//screen in global is used for attenuation
-					Vector2 screen_in_global = to_screen.affine_inverse().xform(screen_size * 0.5);
+	Set<Viewport *> viewports = world_2d->get_viewports();
+	viewports.insert(get_viewport()); // Not sure why this is necessary.
```

Comment:
No idea :P
Is this something that was already done previously, or that you found that was needed for things to work?